src/reflect.ts

```diff
@@ -28,9 +32,85 @@ import { AnyAsyncFunction, AnyFunction } from "./util";
  * @param func an in-line ArrowFunction or FunctionExpression. It must be in-line and cannot reference
  *             a variable or a computed function/closure.
  */
-export declare function reflect<F extends AnyFunction | AnyAsyncFunction>(
+export function reflect<F extends AnyFunction | AnyAsyncFunction>(
   func: F
-): FunctionDecl<F> | Err;
+): FunctionLike<F> | Err | undefined {
+  if (func.name.startsWith("bound ")) {
+    // native bound function
+    const targetFunc = (<any>func)[ReflectionSymbols.TargetFunction];
+    if (targetFunc) {
+      return reflect(targetFunc);
+    } else {
+      throw new Error(
+        `bound function not compiled with Functionless, cannot introspect on its form`
+      );
+    }
+  }
+
+  const astCallback = (<any>func)[ReflectionSymbols.AST];
+  if (typeof astCallback === "function") {
+    if (!reflectCache.has(astCallback)) {
+      reflectCache.set(
+        astCallback,
+        parseSExpr(astCallback()) as FunctionLike | Err
+      );
+    }
+    return reflectCache.get(astCallback) as FunctionLike<F> | Err | undefined;
+  }
+  return undefined;
```

Comment:
What does it mean to return undefined here?

Reply:
It means the function was not registered and cannot be reflected. The caller should handle the error.